network/src/errors.rs

```diff
@@ -18,6 +25,15 @@ pub enum Error {
     Shutdown,
 }
 
+#[derive(Debug)]
+pub enum P2PError {
```

Comment:
Can we just use the `P2PError` in `tentacle`?

Reply:
see https://github.com/nervosnetwork/ckb/pull/2131#discussion_r442120466

Reply:
Got it. I though there was an error enum like this in tentacle but I'm wrong.

@quake What's your opinion? Should P2P define this error enum even it self does not use it in any function?

Reply:
I prefer to let the lib user define the error combinations